first_validation.py

```diff
@@ -48,13 +50,24 @@ def check_hashes(hash_fname):
     # Directory containing hash filenames file.
     data_dir = hash_pth.parent
     # Read in text for hash filename
+    hash_file_contents = hash_pth.read_text()
+
     # Split into lines.
+    lst_of_splitted_lines = hash_file_contents.splitlines()
     # For each line:
+    for line in lst_of_splitted_lines:
         # Split each line into expected_hash and filename
+        [exp_hash, gap, filename] = line.split(sep=' ')
+
         # Calculate actual hash for given filename.
+        act_hash = hash_for_fname(data_dir / filename)
+
         # Check actual hash against expected hash
-        # Return False if any of the hashes do not match.
-    return False
+        result = (act_hash == exp_hash)
```

Comment:
The problem is that you end up returning the result of the _last_ test here, so if any of the previous tests failed, the function still returns True.  I think you want something like:

```
        result = result and (act_hash == exp_hash)
```

to combine the results of the tests.